webapp/redux/middleware/middleware.js

```diff
@@ -106,6 +106,33 @@ const simulateNetwork = (payload) => createSimulateBackendCall(
   GEPPETTO.Resources.RUNNING_SIMULATION,
 );
 
+class ErrorDialogManager {
+  errorTag = 'SimulationId ';
+  errorIds = [];
+  addId(id){ this.errorIds.push(id); }
+  getErrorId(s) {
+    let errorId = 0 ;
+    const i = s.indexOf(this.errorTag);
+    if (i > -1)
+      errorId = parseFloat(s.substring(i + this.errorTag.length, s.length-1));
```

Comment:
the idea of looking into the message and assigning a code based on string finding is hacky and cumbersome. 

If we are ok with string finding I would just implement a function like:

```javascript
shouldLaunchErrorDialog(errorMessage) {
  return !errorMessage.includes(errorTag);
}
```

A better solution would be to format error messages in a serialized format like json, so that the type of the message can be easily parsed.

Reply:
We are not just looking for the tag but for the number (simId) associated to the error message. In the current version we are passing a simId to simulateNetPyNEModelInGeppetto to identify that many exceptions would come from simulate_experiment_trials() in the same experiment run. When the initial dialog is rendered, an entry with the given simId is added to the manager, preventing any further exception to fire the dialog.

On previous discussion with Dario we agreed on using the plain string instead of serialized values.

Reply:
On a second thought, let's use JSON.